File: webapp/server/src/main/java/ecn/tp/bddon/server/metier/dto/Mission.java
package ecn.tp.bddon.server.metier.dto;

import java.io.Serializable;
import java.sql.Timestamp;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import ecn.tp.bddon.server.utils.DateParser;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "mission")
@Data
@NoArgsConstructor
public class Mission implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "mission_id")
    private int id;

    @Column(name = "loading_time")
    private Timestamp loadingTime;

    @ManyToOne
    @JoinColumn(name = "loading_point")
    private LoadingPoint loadingPoint;

    @ManyToOne
    @JoinColumn(name = "driver")
    private Driver driver;

    @ManyToOne
    @JoinColumn(name = "truck")
    private Truck truck;

    public void setLoadingTime(String date) {
        this.loadingTime = DateParser.getTimeStampFromStringDate(date);
    }

}

File: webapp/client/src/app/dto/mission.ts
import { Driver } from "./driver";
import { LoadingPoint } from "./loadingpoint";
import { Truck } from "./truck";

export class Mission {

    static OK = 1;
    static KO = -1;
    static BEGUN = 2;
    static NOT_BEGUN = 0;

    constructor(
        public id: number,
        public loadingTime: Date,
        public loadingPoint: LoadingPoint,
        public driver: Driver,
        public truck: Truck,
        public truckAvailable: boolean,
        public status: number
    ) { }
}